addons/website_sale/models/res_config_settings.py

```diff
@@ -54,23 +54,21 @@ class ResConfigSettings(models.TransientModel):
         related='website_id.show_line_subtotals_tax_selection',
     )
 
-    @api.model
-    def get_values(self):
-        res = super(ResConfigSettings, self).get_values()
-        res.update(
-            enabled_extra_checkout_step=self.env.ref('website_sale.extra_info').active,
-            enabled_buy_now_button=self.env.ref('website_sale.product_buy_now').active,
-        )
-        return res
-
     def set_values(self):
         super().set_values()
-        extra_step_view = self.env.ref('website_sale.extra_info')
-        if extra_step_view.active != self.enabled_extra_checkout_step:
-            extra_step_view.active = self.enabled_extra_checkout_step
-        buy_now_view = self.env.ref('website_sale.product_buy_now')
-        if buy_now_view.active != self.enabled_buy_now_button:
-            buy_now_view.active = self.enabled_buy_now_button
+        extra_step_views = self.with_context(website_id=self.website_id.id).website_id.viewref(
+            'website_sale.extra_info') \
+            if self.website_id else self.env['ir.ui.view'].search([('key', '=', 'website_sale.extra_info')])
```

Comment:
if we find 15 views (eg. 1 global view + 14 for specific websites) we will all archive or unarchive them? it seems a little overreaching. Maybe we could have a search with limit=1 (and probably only search `('website', '=', False)` views) instead of modifying all views, as was done before this change (but not taking into account specific views)

also can't we use viewref without website_id? eg:

```suggestion
        website = self.with_context(website_id=self.website_id.id).website_id
        extra_step_views = website.viewref('website_sale.extra_info')
```
this should give us the view "website_sale.extra_info" for the current website or if there is no website, the default view

Reply:
As we need to synchronize the checkbox in the settings page and the button in the website editor so yes if we get 15 views we need to archive/unarchive them all if you are not choosing a specific website because once we have a specific view for a website the button in the website editor won't be dealing with the general view anymore so archiving and unarchiving the general view at this point won't affect the website and won't show the extra-info page.